packages/account/imAccount/validator.ts

```diff
@@ -1,12 +1,72 @@
-import type { BytesLike, HexString } from "~typing"
+import * as ethers from "ethers"
 
-export enum ValidatorType {
-  ECDSAValidator = "ECDSAValidator"
-}
+import type { BytesLike, HexString } from "~typing"
 
 export interface Validator {
   getAddress(): Promise<HexString>
   getOwnerValidatorInitData(ownerInfo?: string): Promise<HexString>
   getDummySignature(): Promise<HexString>
   sign(message: BytesLike, metadata?: any): Promise<HexString>
 }
+
+export function getValidatorSignMessage(
+  message: BytesLike,
+  validatorAddress: string
+): string {
+  const digest = ethers.keccak256(
+    ethers.AbiCoder.defaultAbiCoder().encode(
+      ["bytes32", "address"],
+      [message, validatorAddress]
+    )
+  )
+
+  return ethers.keccak256(
+    ethers.solidityPacked(
+      ["string", "string", "bytes"],
+      [
+        "\x19Ethereum Signed Message:\n",
+        ethers.dataLength(digest).toString(), // digest length must be 32
+        digest
+      ]
+    )
+  )
+}
+
+export interface WebAuthnValidatorOwner {
+  use(credentialId: string): void
+  sign(
+    challenge: BytesLike,
+    metadata?: any
+  ): Promise<{
+    rawSignature: HexString
+    clientData?: ClientData
+    authenticatorData?: AuthenticatorData
+    webAuthnInput?: WebAuthnInput
```

Comment:
A general owner should return only 

```typescript
{
    signature: HexString
    authenticatorData: string // base64(JSON.stringify({ ...fields }))
    clientData: string // base64(JSON.stringify({ ...fields }))
}
```

And it's validator responsibility to decode those base64 values and get their needing fields.

Reply:
One thing to consider is that should we parse base64 data into object in advance by passkey owner? In some case contract might need plain base64 data, which would cause validator re-calculate the object back to base64 data.